### PR TITLE
Create F2Auth Rockon

### LINKED ISSUE #360: Self hosted 2FA app
With the imminent [Authy for Desktop End of Life EOL](https://support.authy.com/hc/en-us/articles/17592416719003-Authy-for-Desktop-End-of-Life-EOL-) in [EDIT: see next comment] ~~August 2024~~ March 19, 2024 , self hosting option/s such as:

- https://github.com/Bubka/2FAuth - [Docker CLI](https://docs.2fauth.app/getting-started/installation/docker/docker-cli/) - [DockerHub](https://hub.docker.com/r/2fauth/2fauth)

will likely be more in-need.

### PR DESCRIPTION
**Pull request by Hooverdan96**
Fixes #360  .

Note, since this is a draft PR, the root.json has not been updated yet for this PR, in case the direction (aka different docker target) is more viable.

Currently, the installation completes, the container seems to be the running its processes, however the UI screen is blank. Needs further investigation whether it's related to container itself or the Rockon json/other. If anybody has experience with this container/application, please consider helping analyze the issue.

### General information on project
This pull request proposes to add a new rock-on for the following project:
- name: 2FAuth (Desktop authenticator)
- website: https://docs.2fauth.app/
- description: 2FAuth is a web based self-hosted alternative to One Time Passcode (OTP) generators like Google Authenticator, designed for both mobile and desktop.

### Information on docker image
- docker image: https://hub.docker.com/r/2fauth/2fauth
- is an official docker image available for this project?: yes, available for amd64 and arm64

### Checklist
- [X] Passes [JSONlint](https://jsonlint.com) validation
- [x] Entry added to `root.json` in _alphabetical_ order (for new rock-on only)
- [X] `"description"` object lists and links to the docker image used
- [X] `"description"` object provides information on the image's particularities (advantage over another existing rock-on for the same project, for instance)
- [X] `"website"` object links to project's main website

### CODE REVIEW

**Comment by phillxnet** [comment]
Taking a look at this now.

**Review by phillxnet — COMMENTED** [review]
@Hooverdan96 Thanks for stepping up to this Rock-on: much appreciated.

On initial testing I get the following from a local repo!!

requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://build.lan/rockons/F2Auth.json

Just looking into what's not lining up here !!

File: root.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+    "F2Auth": "F2Auth.json",
```
Comment (phillxnet) [inline root.json:2]:
```suggestion
    "2FAuth": "2FAuth.json",
```
@Hooverdan96
This is what I'm hoping we need here :).

**Review by phillxnet — APPROVED** [review]
@Hooverdan96 I've popped in that typo suggestion to complete the review:

As per your comment re name or IP access here: https://github.com/rockstor/rockon-registry/issues/360#issuecomment-1994458463 I used hostname.lan not IP numeric address for both Rockstor Web-UI login and Rock-on config:

2FAuth-all-listing

APP_URL-and-ASSET_URL-config

2FAuth-verify-config

2FAuth-installed

2FAuth-info-page

2FAuth-Web-UI-via-Rock-on-2FAuth_UI-button

So bar the redundant entry of hostname:Web-UI port to get around a current Rock-on limitation, I think we are good on this one. This test was on a lan with no IPv6, using latest testing rpm (5.0.8-0 (RC3)) with Leap 15.5 base OS. As is evident from the last screen pic we are pulling in the latest v5.0.4 of 2FAuth.

I have not tested beyond this point, but intent to merge so we can get wider more informed feedback on this Rock-on.

**Comment by phillxnet** [comment]
PR product PRODUCTION published.

**Comment by Hooverdan96** [comment]
Thanks for the review and catching the typo in the `root.json` file!

**Pull request merged**